awx/main/tests/functional/api/test_workflow_node.py

```diff
@@ -56,6 +68,90 @@ def test_node_accepts_prompted_fields(inventory, project, workflow_job_template,
+        [True, False, 200],
+        [False, False, 403],
+        [False, True, 200],
+    ])
+    def test_approval_node_creation_rbac(self, post, approval_node, alice, is_admin, is_org_admin, status):
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        if is_admin is True:
+            approval_node.workflow_job_template.admin_role.members.add(alice)
+        if is_org_admin is True:
+            approval_node.workflow_job_template.organization.admin_role.members.add(alice)
+        post(url, {'name': 'Test', 'description': 'Approval Node', 'timeout': 0},
+             user=alice, expect=status)
+
+    @pytest.mark.django_db
+    def test_approval_node_exists(self, post, approval_node, admin_user, get):
+        workflow_job_template = WorkflowJobTemplate.objects.create()
+        approval_node = WorkflowJobTemplateNode.objects.create(
+            workflow_job_template=workflow_job_template
+        )
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        post(url, {'name': 'URL Test', 'description': 'An approval', 'timeout': 0},
+             user=admin_user)
+        get(url, admin_user, expect=200)
+
+    @pytest.mark.django_db
+    def test_activity_stream_create_wf_approval(self, post, admin_user, workflow_job_template):
+        wfjn = WorkflowJobTemplateNode.objects.create(workflow_job_template=workflow_job_template)
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': wfjn.pk, 'version': 'v2'})
+        post(url, {'name': 'Activity Stream Test', 'description': 'Approval Node', 'timeout': 0},
+             user=admin_user)
+
+        qs1 = ActivityStream.objects.filter(organization__isnull=False)
+        assert qs1.count() == 1
+        assert qs1[0].operation == 'create'
+
+        qs2 = ActivityStream.objects.filter(organization__isnull=True)
+        assert qs2.count() == 5
+        assert qs2[0].operation == 'create'
+        assert qs2[1].operation == 'create'
+        assert qs2[2].operation == 'create'
+        assert qs2[3].operation == 'create'
+        assert qs2[4].operation == 'update'
+
+    def test_approval_node_cleanup(self, post, approval_node, admin_user, get):
+        workflow_job_template = WorkflowJobTemplate.objects.create()
+        approval_node = WorkflowJobTemplateNode.objects.create(
+            workflow_job_template=workflow_job_template
+        )
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+
+        post(url, {'name': 'URL Test', 'description': 'An approval', 'timeout': 0},
+             user=admin_user)
+        workflow_job_template.delete()
+        get(url, admin_user, expect=404)
```

Comment:
In this situation, `url` is URL of the approval endpoint i.e., `/api/v2/workflow_job_template_nodes/N/create_approval/`, and this is going to return a 404 because the node got deleted.

But we *also* want to verify that the `WorkflowApprovalTemplate` itself got deleted, and you could do that *without* having to use the `get` fixture:

```python
assert WorkflowApprovalTemplate.objects.count() == 1
workflow_job_template.delete()
assert WorkflowApprovalTemplate.objects.count() == 0
```